src/Microsoft.Azure.Functions.Extensions.Mcp/Http/McpHttpUtility.cs

```diff
@@ -39,4 +42,82 @@ internal static void SetSseContext(HttpContext context)
 
         context.Features.GetRequiredFeature<IHttpResponseBodyFeature>().DisableBuffering();
     }
+
+    internal static async ValueTask<JsonRpcMessage?> ProcessJsonRpcPayloadAsync(HttpRequest request, JsonSerializerOptions options, CancellationToken cancellationToken, bool unwrapOnly = false)
+    {
+        // Process the incoming request body as JSON. Support both raw JSON-RPC messages and
+        // wrapped payloads with the shape: { "isFunctionsMcpResult": true, "content": <JSON-RPC message> }
+        // 
+        // When unwrapOnly is false: If the wrapper is present, deserialize the inner "content" as the JsonRpcMessage. 
+        // Otherwise, deserialize the root object directly as a JsonRpcMessage and return it.
+        //
+        // When unwrapOnly is true: If the wrapper is present, replace the request.Body stream with a memory stream 
+        // containing only the inner content. If no wrapper is present, leave the original body intact.
+
+        // If the body is empty, return null.
+        if (request.ContentLength == null || request.ContentLength == 0)
+        {
+            return null;
+        }
+
+        // Read the request body into a JsonDocument for inspection.
+        request.EnableBuffering();
```

Comment:
Could another component have read the body before we get to this part of the code? If so, do we need to reset the position?

Reply:
It might be safer to always reset